syscalls.c

```diff
@@ -1281,12 +1549,25 @@ ssize_t syscalls_sys_recvmsg(char *ustack)
 	GETFROMSTACK(ustack, struct msghdr *, msg, 1);
 	GETFROMSTACK(ustack, int, flags, 2);
 
+	if (vm_mapBelongs(proc, msg, sizeof(*msg)) < 0) {
+		return -EFAULT;
+	}
+
+	if (vm_mapBelongs(proc, msg->msg_iov, sizeof(msg->msg_iovlen)) < 0) {
```

Comment:
From posix:
> `msg_iov` points to an array of `iovec` structures; `msg_iovlen` must be set to the dimension of this array
```suggestion
	if (vm_mapBelongs(proc, msg->msg_iov, sizeof(*msg->msg_iov) * msg->msg_iovlen) < 0) {
```

And there's more:
> In each `iovec` structure, the `iov_base` field specifies a storage area and the `iov_len` field gives its size in bytes. Some of these sizes can be zero.